src/main/scala/com/redis/sentinel/SentinelMonitor.scala

```diff
@@ -111,13 +118,17 @@ class SentinelMonitor (address: SentinelAddress, listener: SentinelListener, con
 
 trait SentinelHeartBeater extends Runnable with Log{
   private var running = false
+  private val beatingSentinel = new AtomicBoolean(true) // initialised to true so we don't unnecessarily wake up ops
```

Comment:
So. We could use a normal boolean, but it'd have to be marked volatile. Since that code is almost certainly not in the hot code path, the performance gain is negligible. I do believe, however, that without volatile, the code would be incorrect.